docs/DeveloperGuide.md

```diff
@@ -671,29 +673,416 @@ testers are expected to do more *exploratory* testing.
+
+    3. Test case: `unmarko 0`<br>
+       Expected: No order is reverted to pending status. Error details shown in the status message. Status bar remains the same.
+
+3. Reverting an order which is already pending to pending.
+
+    1. Prerequisites:
+        - Prerequisites: At least 1 order listed. <br>
+        - The first order's status is `pending`.
+
+    2. Test case: `unmarko 1`<br>
+       Expected: No order is reverted to pending status. Error details shown in the status message. Status bar remains the same.
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+    1. Prerequisite: You have not edited the `preferences.json` file. There is a folder named `data` in the same directory as the jar file, and there is a `addresbook.json` file in the `data` folder.
+
+    2. Test case: Delete the `addresbook.json` file. Then, run SellSavvy and exit using the `exit` command.<br>
+       Expected: SellSavvy should create a new `addresbook.json` file with default data.
+
+    3. Test case: Delete the `data` folder together with the `addresbook.json` file. Then, run SellSavvy and exit using the `exit` command.<br>
+       Expected: SellSavvy should create a new `data` folder and a new `addresbook.json` file inside the folder with default data.
+
+    4. Test case: Corrupt the `addressbook.json` file by changing its contents to an invalid format, e.g., add a non-numeric character to one of the customer's phone number. Then, run SellSavvy and exit using the `exit` command.<br>
+       Expected: SellSavvy should discard all data in the file and start with an `addressbook.json` file with an empty person list.
+
+## **Appendix: Planned Enhancements**
+
+**Team size: 4**
+
+1. **Change the `CUSTOMER_INDEX` of `addorder` command to be optional when a customer's order list is open** <br>
+    - **Problem:** Currently, we made the customer's index needed to be specified for `addorder` when the customer's order list is open but related commands like `editorder` and `filterorder` do not require a customer's index to be supplied.
+    - **Solution:** We planned to make the customer's index for `addorder` optional when a particular customer's order list is already displayed.
+    - **Rationale:** Requiring the user to remember the customer's index seems unnecessary.
 
-1. _{ more test cases …​ }_
+2. **Allows detection of wrong prefix for commands**
+    - **Problem:** Currently, we only detect the relevant prefix for each command and treat every other prefix as part of the input.
+    - **Example:** User tries to edit an existing order using `edito 1 d/01-12-2023 n/item`. An error message informs user that date is wrong.
+    - **Solution:** Add the functionality to detect such errors and inform the user of the wrong prefix used instead of treating it as part of the parameters.
```

Comment:
Can add what was suggested regarding a few exceptions for parameters such as item name?
e.g. in cases where they want to add `n/` into an item name.

Reply:
done.